modules/chat/src/Chat.ts

```diff
@@ -1,47 +1,66 @@
-import ConduitGrpcSdk, {
-  ConduitServiceModule,
+import {
+  ManagedModule,
   DatabaseProvider,
-  GrpcServer,
-  SetConfigRequest,
-  SetConfigResponse,
-} from '@conduitplatform/grpc-sdk';
-import ChatConfigSchema from './config';
-import { status } from '@grpc/grpc-js';
-import path from 'path';
-import { isArray, isNil } from 'lodash';
+} from "@conduitplatform/grpc-sdk";
+
+import AppConfigSchema from './config';
+import { AdminHandlers } from './admin/admin';
 import { ChatRoutes } from './routes/routes';
 import * as models from './models';
-import { ChatMessage, ChatRoom } from './models';
-import { AdminHandlers } from './admin/admin';
 import { validateUsersInput } from './utils';
+import { ChatMessage, ChatRoom } from './models';
+import ChatConfigSchema from './config';
+import path from 'path';
+import { isArray, isNil } from 'lodash';
+import { status } from '@grpc/grpc-js';
 
-export default class ChatModule extends ConduitServiceModule {
-  private database: DatabaseProvider;
-  private _admin: AdminHandlers;
+export default class Authentication extends ManagedModule {
+  config = AppConfigSchema;
+  service = {
+    protoPath: path.resolve(__dirname, 'chat.proto'),
+    protoDescription: 'chat.Chat',
+    functions: {
+      setConfig: this.setConfig.bind(this),
+      createRoom: this.createRoom.bind(this),
+      deleteRoom: this.deleteRoom.bind(this),
+      sendMessage: this.sendMessage.bind(this),
+    },
+  };
   private isRunning: boolean = false;
-  private _router: ChatRoutes;
+  private adminRouter: AdminHandlers;
+  private userRouter: ChatRoutes;
+  private database: DatabaseProvider;
 
-  constructor(grpcSdk: ConduitGrpcSdk) {
-    super();
-    this.grpcSdk = grpcSdk;
+  constructor() {
+    super('chat');
   }
 
-  async initialize(servicePort?: string) {
-    this.grpcServer = new GrpcServer(servicePort);
-    this._port = (await this.grpcServer.createNewServer()).toString();
-    await this.grpcServer.addService(
-      path.resolve(__dirname, './chat.proto'),
-      'chat.Chat',
-      {
-        setConfig: this.setConfig.bind(this),
-        createRoom: this.createRoom.bind(this),
-        deleteRoom: this.deleteRoom.bind(this),
-        sendMessage: this.sendMessage.bind(this),
+  async onServerStart() {
+    await this.grpcSdk.waitForExistence('database');
+    this.database = this.grpcSdk.databaseProvider!;
+  }
+
+  protected registerSchemas() {
+    const promises = Object.values(models).map((model: any) => {
+      const modelInstance = model.getInstance(this.database);
+      if (Object.keys(modelInstance.fields).length !== 0) { // borrowed foreign model
+        return this.database.createSchemaFromAdapter(modelInstance);
       }
-    );
-    await this.grpcServer.start();
+    });
+    return Promise.all(promises);
+  }
+
+  async onConfig() {
```

Comment:
this should just call enableModule, why would it be duplicated?

Reply:
mb, leftover code